File: Java/0051-0100/75. Sort Colors.java
// https://leetcode.com/problems/sort-colors/

class Solution {
    public void sortColors(int[] nums) {
        int[] count = new int[3];

        for(int num : nums) count[num]++; // Count the occurance of 0,1 & 2
        
        int index = 0;
        for(int value=0; value<3; value++)
            for(int occurance=0; occurance<count[value]; occurance++)
                nums[index++] = value;
    }
}

File: Javascript/0051-0100/75. Sort Colors.js
// https://leetcode.com/problems/sort-colors/

var sortColors = function(nums) {
    const count = [0,0,0];

    for(let num of nums) count[num]++; // Count the occurance of 0,1 & 2
    
    let index = 0;
    for(let value=0; value<3; value++)
        for(let occurance=0; occurance<count[value]; occurance++)
            nums[index++] = value;
};
